cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,172 @@
+            ]
+        instancerole = iam.CfnRole(
+            self,
+            id="InstanceRole",
+            managed_policy_arns=managed_policy_arns,
+            assume_role_policy_document={
+                "Statement": {
+                    "Action": "sts:AssumeRole",
+                    "Effect": "Allow",
+                    "Principal": {"Service": "ec2.amazonaws.com"},
+                },
+                "Version": "2012-10-17",
+            },
+            path="/executionServiceEC2Role/",
+        )
+        instancerole.add_metadata("Comment", "Role to be used by instance during image build.")
+
+        # InstanceProfile
+        iam.CfnInstanceProfile(
+            self, id="InstanceProfile", path="/executionServiceEC2Role/", roles=[core.Fn.ref("InstanceRole")]
+        )
+
+        # InfrastructureConfiguration
+        imagebuilder.CfnInfrastructureConfiguration(
+            self,
+            id="PClusterImageInfrastructureConfiguration",
+            name="PCluster-Image-Infrastructure-Configuration",
+            instance_profile_name=core.Fn.ref("InstanceProfile"),
+            terminate_instance_on_failure=build_config.terminate_on_failure,
+            instance_types=instance_types,
+        )
+
+        # Define ami build instance ebs
+        ebs = imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
+            delete_on_termination=image_config.root_volume.delete_on_termination,
+            volume_size=image_config.root_volume.size,
+            volume_type="gp2",
+        )
+
+        cloudformation_dir = "../../../../cloudformation/imagebuilder/"
+
+        # Component
+        if dev_settings_config and dev_settings_config.update_and_reboot:
+            imagebuilder.CfnComponent(
+                self,
+                id="UpdateAndRebootComponent",
+                name="UpdateAndReboot",
+                version="0.0.1",
+                description="Update OS and Reboot",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "update_and_reboot.yaml")),
+            )
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster AMI",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "pcluster_install.yaml")),
+            )
+
+            components = [
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("UpdateAndReboot")
+                ),
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                ),
+            ]
+        else:
```

Comment:
can we factor out the PClusterComponent, removing the else?
something like
```
components = []

if update_and_reboot:
    create UpdateAndRebootComponent
    components += [UpdateAndRebootComponent]

create PClusterComponent
components += [PClusterComponent]
```

Reply:
Done